snapcraft/internal/build_providers/_multipass/_multipass_command.py

```diff
@@ -304,21 +327,86 @@ def umount(self, *, mount: str) -> None:
                 provider_name=self.provider_name, exit_code=process_error.returncode
             ) from process_error
 
-    def copy_files(self, *, source: str, destination: str) -> None:
-        """Passthrough for running multipass copy-files.
-
-        :param str source: the source file to copy, using syntax expected
-                           by multipass.
-        :param str destination: the destination of the copied file, using
-                                syntax expected by multipass.
+    def copy_files(  # noqa: C901
+        self,
+        *,
+        source: Union[str, IO],
+        destination: Union[str, IO],
+        bufsize: int = 1024
+    ) -> None:
+        """Passthrough for running multipass copy-files. Only one of source and
+        destination may be a stream.
+
+        :param str or IO source: the source file to copy, using syntax expected
+                                 by multipass, or a file-like object to read from
+        :param str or IO destination: the destination of the copied file, using
+                                      syntax expected by multipass, or a file-like
+                                      object to write to
         """
-        cmd = [self.provider_cmd, "copy-files", source, destination]
-        try:
-            _run(cmd)
-        except subprocess.CalledProcessError as process_error:
+        # can't use std{in,out}=open(...) due to LP#1849753
+        args = [self.provider_cmd, "transfer"]
+        kwargs = {}  # type: Dict[str, str]
```

Comment:
The best syntax for mypy annotations is now:
`<var>: <annotation> = foo`

i.e.
`kwargs: Dict[str, str] = {}`

Reply:
I would suggest instead of packing arguments into the dictionary, keep them as variables w/ defaults.
```
stdin = None
stdout = None
...
   stdin = subprocess.PIPE
...
_popen(args, stdin=stdin, stdout=stdout)
...
```

Reply:
Neither of these applies any more.